torchmetrics/functional/image/sam.py

```diff
@@ -62,20 +67,22 @@ def _sam_compute(
             - ``'none'``: no reduction will be applied
 
     Example:
-        >>> preds = torch.rand([16, 1, 16, 16], generator=torch.manual_seed(42))
-        >>> target = torch.rand([16, 1, 16, 16], generator=torch.manual_seed(123))
+        >>> preds = torch.rand([16, 3, 16, 16], generator=torch.manual_seed(42))
+        >>> target = torch.rand([16, 3, 16, 16], generator=torch.manual_seed(123))
         >>> preds, target = _sam_update(preds, target)
         >>> _sam_compute(preds, target)
-        tensor(0.7377)
+        tensor(0.5943)
     """
     B, C, H, W = preds.shape
     preds = preds.reshape(B, C, H * W)
+    preds = torch.movedim(preds, 1, -1)
     target = target.reshape(B, C, H * W)
+    target = torch.movedim(target, 1, -1)
```

Comment:
Thank you for your suggestion. I agree it's better to return `(B, H, W)` than `(B, H * W)`, it will make the output more flexible with different use cases